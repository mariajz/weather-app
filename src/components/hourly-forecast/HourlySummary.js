import React from 'react';
import { StyledText } from '../../commons/styles';
import { degreeSymbol } from '../../constants/Constants';
import { CustomImage } from '../../commons/visual-elements';
import { HourlySummaryWrapper } from './HourlyForecast.style';

const HourlySummary = ({ hourlyWeatherItem }) => {
    const { weatherImage, time, temp } = hourlyWeatherItem;

    return (
        <>
            <HourlySummaryWrapper>
                <StyledText size={10}>{time}</StyledText>
                <CustomImage path={weatherImage} width={25} height={25} />
                <StyledText size={10}>
                    {temp}
                    {degreeSymbol}C
                </StyledText>
            </HourlySummaryWrapper>
        </>
    );
};

export default HourlySummary;
