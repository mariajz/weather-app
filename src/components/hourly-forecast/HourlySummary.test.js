import { render } from '@testing-library/react-native';
import React from 'react';
import HourlySummary from './HourlySummary';

const defaultProps = {
    hourlyWeatherItem: {
        condition: 'Overcast',
        temp: 23.3,
        time: '11 PM',
        weatherImage: 11,
    },
};
describe('HourlySummary', () => {
    it('should render HourlySummary', () => {
        const container = render(<HourlySummary {...defaultProps} />);

        expect(container).toMatchSnapshot();
    });
});
