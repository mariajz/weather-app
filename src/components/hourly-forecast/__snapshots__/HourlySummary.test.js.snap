// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HourlySummary should render HourlySummary 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "column",
      "height": 60,
      "justifyContent": "space-between",
      "maxWidth": 60,
      "paddingBottom": 0,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 0,
    }
  }
>
  <Text
    size={10}
    style={
      {
        "color": "white",
        "fontSize": 10,
        "marginLeft": 0,
      }
    }
  >
    11 PM
  </Text>
  <Image
    height={25}
    source={11}
    style={
      {
        "height": 25,
        "width": 25,
      }
    }
    testID="image"
    width={25}
  />
  <Text
    size={10}
    style={
      {
        "color": "white",
        "fontSize": 10,
        "marginLeft": 0,
      }
    }
  >
    23.3
    Â°
    C
  </Text>
</View>
`;
